src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -1383,23 +1457,41 @@ function ScatterplotWithControls({
             }}
           >
             <Switch
-              label="Log Scale:"
+              label="Log scale (will exclude values &le; 0):"
               state={vizConfig.independentAxisLogScale}
-              onStateChange={onIndependentAxisLogScaleChange}
+              onStateChange={(newValue: boolean) => {
+                setDismissedIndependentAllNegativeWarning(false);
+                onIndependentAxisLogScaleChange(newValue);
+              }}
               // disable log scale for date variable
-              disabled={independentValueType === 'date'}
+              disabled={independentValueType === 'date' || valueSpec != 'Raw'}
             />
           </div>
+          {independentAllNegative && !dismissedIndependentAllNegativeWarning ? (
+            <Notification
+              title={''}
+              text={
+                'Nothing can be plotted with log scale because all values are negative or zero'
+              }
+              color={'#5586BE'}
```

Comment:
should probably get the warning color from CoreUI. Can be left for another ticket though!

Reply:
Yes, I'm happy to search for them with some regexp magic when we do this (rather than make a ticket for this specifically)